### PR TITLE
Create Password Expiry Notification Using Teams and Graph API

### PR DESCRIPTION
**Pull request by farismalaeb**
# PR Summary

## PR Checklist

- Do you want this post to be published on a specific date?: Y/N [Specify date]:
- [x] I have read the [contributors guide][contrib] and followed the style and process guidelines
- [x] PR has a meaningful title
- [x] Includes content related an open issue
  - see [Closing issues using keywords][key].
- [x] This PR is ready to merge and is not **Work in Progress**

[contrib]: https://docs.microsoft.com/powershell/scripting/community/contributing/overview
[key]: https://help.github.com/en/articles/closing-issues-using-keywords

### CODE REVIEW

File: Posts/2022/07/Password Expiry Notification Using Teams and Graph API [hunk 1]
```diff
@@ -0,0 +1,277 @@
+---
+post_title: Password Expiry Notification Using Teams and Graph API
+username: farismalaeb
+categories: PowerShell
+tags: GraphAPI, Teams, PowerShell
+summary: This post's intent is to show how to use Graph API through PowerShell to send a Teams message.
+---
+
+**Q**: How to send a password expiration notification to a user using Teams chat.
```
Comment (michaeltlombardi) [inline Posts/2022/07/Password Expiry Notification Using Teams and Graph API:9]:
```suggestion
**Q**: How do I send a password expiration notification to a user using Teams chat?
```

File: Posts/2022/07/Password Expiry Notification Using Teams and Graph API [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+
+**A**: It's not only the password notification, but you can PowerShell and invoke the Teams Graph API to send any message to Teams user.
+But first, let's talk about Graph API, so we are all on the same page.
```
Comment (michaeltlombardi) [inline Posts/2022/07/Password Expiry Notification Using Teams and Graph API:12]:
```suggestion
**A**: Not only can you send the password notification, but you can use PowerShell with the Teams Graph API to send any message to a Teams user.

But first, let's talk about Graph API, so we are all on the same page.
```

File: Posts/2022/07/Password Expiry Notification Using Teams and Graph API [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+[Microsoft Graph](https://docs.microsoft.com/en-us/graph/overview) is the gateway to read data from a wide range of Microsoft services, including Azure Active Directory, Teams, OneDrive…etc.
+You can get the data using a single module and a single interface.
+
+Microsoft Graph API supports modern authentication protocols such as access token, certificate, and browser authentication.
+
+> You can read more about the Graph API available endpoint from the [Microsoft Graph REST API Endpoint v1.0 Reference](https://docs.microsoft.com/en-us/graph/api/overview?view=graph-rest-1.0&preserve-view=true).
+
+## Downloading Graph API PowerShell Module
+
+You can download Microsoft Graph PowerShell Module by running the following line.
```
Comment (michaeltlombardi) [inline Posts/2022/07/Password Expiry Notification Using Teams and Graph API:29]:
```suggestion
You can download Microsoft Graph PowerShell Module by running the following command.
```

File: Posts/2022/07/Password Expiry Notification Using Teams and Graph API [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+```powershell
+Install-Module -Name Microsoft.Graph
+```
+
+Microsoft PowerShell Graph Module SDK is cross-platform and supports Windows, macOS, and Linux.
+
+## Connecting To Graph API Using PowerShell
+
+Unlike other modules such as the AzureAD, ExchangeOnline…etc.
+were the admin needs only to connect with the right credentials and have full access, the graph has a different approach.
```
Comment (michaeltlombardi) [inline Posts/2022/07/Password Expiry Notification Using Teams and Graph API:40]:
```suggestion
where the admin needs only to connect with the right credentials and have full access, the graph has a different approach.  
```

File: Posts/2022/07/Password Expiry Notification Using Teams and Graph API [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+## Connecting To Graph API Using PowerShell
+
+Unlike other modules such as the AzureAD, ExchangeOnline…etc.
+were the admin needs only to connect with the right credentials and have full access, the graph has a different approach.  
+
+When connecting to the Graph API, you need to specify the scope of permissions or, let's say, declare the required permissions that are used during the script execution.
+The script fails if it tries to perform an action that was not in the scope.
+
+For example, if the script needs to read all user data in the azure directory, it's not enough just to connect to read all the data, even if the user credentials are for the global admin for the tenant.
+Instead, you must declare and specify that you will connect and use the **User.Read.All.**
```
Comment (michaeltlombardi) [inline Posts/2022/07/Password Expiry Notification Using Teams and Graph API:46]:
```suggestion
Instead, you must declare and specify that you will connect and use the `User.Read.All` permission.
```

File: Posts/2022/07/Password Expiry Notification Using Teams and Graph API [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+To connect to Graph API with the required scope, use the following:
+
+```powershell-console
+PS> $Scope=@('User.Read.All','User.ReadWrite.All')
+PS> Connect-MgGraph -Scopes $Scope
+
+Welcome To Microsoft Graph!
+```
+
+To check which identity is used during the connecting with the Graph API along with the used scope, use the `Get-MgContext`
```
Comment (michaeltlombardi) [inline Posts/2022/07/Password Expiry Notification Using Teams and Graph API:57]:
```suggestion
To check which identity is used during the connecting with the Graph API along with the used scope, use the `Get-MgContext` cmdlet.
```

File: Posts/2022/07/Password Expiry Notification Using Teams and Graph API [hunk 7]
```diff
@@ -0,0 +1,277 @@
...
+CertificateName       : 
+Account               : farismalaeb@contoso.com
+AppName               : Microsoft Graph PowerShell
+ContextScope          : CurrentUser
+Certificate           : 
+PSHostVersion         : 2022.6.3
+ClientTimeout         : 00:05:00
+```
+
+To add additional permission to the scope, rerun the `Connect-MgGraph`, set the new scope to the **Scope** parameter and connect again. There is no need to specify the same scope already provided.
```
Comment (michaeltlombardi) [inline Posts/2022/07/Password Expiry Notification Using Teams and Graph API:77]:
```suggestion
To add additional permission to the scope, rerun `Connect-MgGraph`, setting the new scope with the **Scope** parameter and connect again. There is no need to specify the same scope already provided.
```

File: Posts/2022/07/Password Expiry Notification Using Teams and Graph API [hunk 8]
```diff
@@ -0,0 +1,277 @@
...
+
+So let's assume that an admin connected to the Graph API using the following scope
```
Comment (michaeltlombardi) [inline Posts/2022/07/Password Expiry Notification Using Teams and Graph API:79]:
```suggestion
So let's assume that an admin connected to the Graph API using the following scope:
```

File: Posts/2022/07/Password Expiry Notification Using Teams and Graph API [hunk 9]
```diff
@@ -0,0 +1,277 @@
...
+
+```powershell
+$Scope=@('User.Read.All')
+Connect-MgGraph -Scopes $Scope
+```
+
+Later on, the admin wants to add the **User.ReadWrite.All** permission, all the admin needs to do is run the `Connect-MgGraph` and set the new scope
```
Comment (michaeltlombardi) [inline Posts/2022/07/Password Expiry Notification Using Teams and Graph API:86]:
```suggestion
Later on, the admin wants to add the `User.ReadWrite.All` permission, all the admin needs to do is run the `Connect-MgGraph` and set the new scope
```

File: Posts/2022/07/Password Expiry Notification Using Teams and Graph API [hunk 10]
```diff
@@ -0,0 +1,277 @@
...
+```powershell
+$Scope=@('User.ReadWrite.All')
+Connect-MgGraph -Scopes $Scope
+```
+
+The new scope permission adds to the current one.
+
+A good starting point in finding out the required permission to execute a certain cmdlet is [Find-MgGraphcommand](https://docs.microsoft.com/en-us/powershell/microsoftgraph/find-mg-graph-command?view=graph-powershell-1.0) and [Graph Explorer](https://developer.microsoft.com/en-us/graph/graph-explorer).
+
+## **Password Expiry Notification Using Teams and Graph API**
```
Comment (michaeltlombardi) [inline Posts/2022/07/Password Expiry Notification Using Teams and Graph API:97]:
```suggestion
## Password Expiry Notification Using Teams and Graph API
```

File: Posts/2022/07/Password Expiry Notification Using Teams and Graph API [hunk 11]
```diff
@@ -0,0 +1,277 @@
...
+
+### Getting Password Expiration information
+
+To make this tutorial more fun, let's make a scenario.
+Consider a company named Contoso.com.
+Contoso.com have an on-premise AD syncing with AAD. The Password Expiration policy is set to 3 months.
+
+The administrator is looking for a way to send the users a notification through Microsoft Teams chat one week before the password expires, so how to start.
```
Comment (michaeltlombardi) [inline Posts/2022/07/Password Expiry Notification Using Teams and Graph API:105]:
```suggestion
The administrator is looking for a way to send the users a notification through Microsoft Teams chat one week before the password expires, so, how to start?
```

File: Posts/2022/07/Password Expiry Notification Using Teams and Graph API [hunk 12]
```diff
@@ -0,0 +1,277 @@
...
+
+To get the password expiration for users, use the following code.
+This code reads the **Name**, **EmailAddress**, **UserPrincipalName** and **[msDS-UserPasswordExpiryTimeComputed](https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-adts/f9e9b7e2-c7ac-4db6-ba38-71d9696981e9)**.
```
Comment (michaeltlombardi) [inline Posts/2022/07/Password Expiry Notification Using Teams and Graph API:108]:
```suggestion
This code reads the **Name**, **EmailAddress**, **UserPrincipalName** and [**msDS-UserPasswordExpiryTimeComputed**](https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-adts/f9e9b7e2-c7ac-4db6-ba38-71d9696981e9).
```

File: Posts/2022/07/Password Expiry Notification Using Teams and Graph API [hunk 13]
```diff
@@ -0,0 +1,277 @@
...
+The **msDS-UserPasswordExpiryTimeComputed** return when the users password expires, check it below.
```
Comment (michaeltlombardi) [inline Posts/2022/07/Password Expiry Notification Using Teams and Graph API:109]:
```suggestion
The **msDS-UserPasswordExpiryTimeComputed** property notes when the user's password expires, check it below.
```

File: Posts/2022/07/Password Expiry Notification Using Teams and Graph API [hunk 14]
```diff
@@ -0,0 +1,277 @@
...
+
+```powershell
+$DaysToSendWarning = (Get-Date).AddDays(7).ToLongDateString()
+**$Users = Get-ADUser -Filter {Enabled -eq $True -and PasswordNeverExpires -eq $False -and PasswordLastSet -gt 0 } `
+    -Properties "Name", "EmailAddress", "msDS-UserPasswordExpiryTimeComputed","UserPrincipalName" -SearchBase $LDAPdistinguishedName | Select-Object -Property "Name", "UserPrincipalName","EmailAddress", `
+    @{Name = "PasswordExpiry"; Expression = {[datetime]::FromFileTime($_."msDS-UserPasswordExpiryTimeComputed").ToLongDateString() }}**
```
Comment (michaeltlombardi) [inline Posts/2022/07/Password Expiry Notification Using Teams and Graph API:115]: Reworked things a little to use splatting/limit line lengths as much as possible for readers.

```suggestion
$DaysToSendWarning = (Get-Date).AddDays(7).ToLongDateString()

$QueryParameters = @{
    Filter     = {
        Enabled -eq $true -and
        PasswordNeverExpires -eq $false -and
        PasswordLastSet -gt 0
    }
    Properties = @(
        'Name'
        'EmailAddress'
        'msDS-UserPasswordExpiryTimeComputed'
        'UserPrincipalName'
    )
    SearchBase = $LDAPdistinguishedName
}

$SelectionProperties = @(
    "Name"
    "UserPrincipalName"
    "EmailAddress"
    @{
        Name = 'PasswordExpiry'
        Expression = {
            [datetime]::FromFileTime($_."msDS-UserPasswordExpiryTimeComputed").ToLongDateString()
        }
    }
)

$Users = Get-ADUser @QueryParameters | Select-Object -Property $SelectionProperties
```

File: Posts/2022/07/Password Expiry Notification Using Teams and Graph API [hunk 15]
```diff
@@ -0,0 +1,277 @@
...
+```
+
+To see the result, call the **$Users** variable.
```
Comment (michaeltlombardi) [inline Posts/2022/07/Password Expiry Notification Using Teams and Graph API:118]:
```suggestion
To see the result, call the `$Users` variable.
```

File: Posts/2022/07/Password Expiry Notification Using Teams and Graph API [hunk 16]
```diff
@@ -0,0 +1,277 @@
...
+test               test@contoso.com          test1@contoso.com      Monday, August 15, 2022
+User1              User1@contoso.com         User1@contoso.com      Tuesday, October 18, 2022
+User2              User2@contoso.com         User2@contoso.com      Sunday, August 7, 2022
+.
+.
+.
+Output trimmed
+```
+
+Later on a `ForEach-Object` loop goes through the users, and if any user **msDS-UserPasswordExpiryTimeComputed** match the date in the **$DaysToSendWarning,**, then the script sends them the chat message.
```
Comment (michaeltlombardi) [inline Posts/2022/07/Password Expiry Notification Using Teams and Graph API:132]:
```suggestion
Later on a `ForEach-Object` loop goes through the users, and if any user **msDS-UserPasswordExpiryTimeComputed** matches the date in the `$DaysToSendWarning`, then the script sends them the chat message.
```

File: Posts/2022/07/Password Expiry Notification Using Teams and Graph API [hunk 17]
```diff
@@ -0,0 +1,277 @@
...
+                        @{
+                            "@odata.type" = "#microsoft.graph.aadUserConversationMember"
+                            Roles = @(
+                                "owner"
+                            )
+                            "User@odata.bind" = "https://graph.microsoft.com/v1.0/users('*XXXXXXXXXX*')"
+                        }
+                    )
+                }
+                $ChatSessionID=New-MgChat -BodyParameter $NewChatIDParam
```
Comment (michaeltlombardi) [inline Posts/2022/07/Password Expiry Notification Using Teams and Graph API:168]: Slight reformatting for line length.
```suggestion
$NewChatIDParam = @{
    ChatType = "oneOnOne"
    Members  = @(
        @{
            "@odata.type" = "#microsoft.graph.aadUserConversationMember"
            Roles = @(
                "owner"
            )
            "User@odata.bind" = "https://graph.microsoft.com/v1.0/users('*XXXXXXXXXX*')"
        }
        @{
            "@odata.type" = "#microsoft.graph.aadUserConversationMember"
            Roles = @(
                "owner"
            )
            "User@odata.bind" = "https://graph.microsoft.com/v1.0/users('*XXXXXXXXXX*')"
        }
    )
}

$ChatSessionID = New-MgChat -BodyParameter $NewChatIDParam
```

File: Posts/2022/07/Password Expiry Notification Using Teams and Graph API [hunk 18]
```diff
@@ -0,0 +1,277 @@
...
+
+
+PS> $ChatSessionID.id
+19:b980153c-9129-9129-9129-fb57a348d4d3_eafa5e65-9129-4e69-4e69-19f3fe6@unq.gbl.spaces
+
+```
+
+Replace each *XXXXXXXXXX* with a user ID who is participating in the chat. It doesn't matter if the sender or the recipient is first. it's a two-way communication bridge.
```
Comment (michaeltlombardi) [inline Posts/2022/07/Password Expiry Notification Using Teams and Graph API:176]:
```suggestion
Replace each `*XXXXXXXXXX*` with a user ID who is participating in the chat. It doesn't matter if the sender or the recipient is first. it's a two-way communication bridge.
```

File: Posts/2022/07/Password Expiry Notification Using Teams and Graph API [hunk 19]
```diff
@@ -0,0 +1,277 @@
...
+              }
+              
+        New-MgChatMessage -ChatId $ChatSessionID.ID -Body $Body -Importance Urgent 
+        }
+        catch{
+            write-host $_.Exception.Message
+        }
+ }
+
+}
```
Comment (michaeltlombardi) [inline Posts/2022/07/Password Expiry Notification Using Teams and Graph API:267]: Some minor changes for readability/line length
```suggestion
Import-Module ActiveDirectory
Import-Module Microsoft.Graph.Teams

$Scope = @(
    'Chat.Create'
    'Chat.ReadWrite'
    'User.Read'
    'User.Read.All'
)
Connect-MgGraph -Scopes $Scope

$DaysToSendWarning = 7

#Find accounts that are enabled and have expiring passwords
$QueryParameters = @{
    Filter     = {
        Enabled -eq $true -and
        PasswordNeverExpires -eq $false -and
        PasswordLastSet -gt 0
    }
    Properties = @(
        'Name'
        'EmailAddress'
        'msDS-UserPasswordExpiryTimeComputed'
        'UserPrincipalName'
    )
    SearchBase = $LDAPdistinguishedName
}

$SelectionProperties = @(
    "Name"
    "UserPrincipalName"
    "EmailAddress"
    @{
        Name = 'PasswordExpiry'
        Expression = {
            [datetime]::FromFileTime($_."msDS-UserPasswordExpiryTimeComputed").ToLongDateString()
        }
    }
)

$Users = Get-ADUser @QueryParameters | Select-Object -Property $SelectionProperties

foreach ($User in $Users) {
    $RecpID = Get-MgUser -UserId $User.UserPrincipalName -ErrorAction Stop
    if ($User.PasswordExpiry -eq $DaysToSendWarning) {
        $NewChatIDParam = @{
            ChatType = "oneOnOne"
            Members = @(
                @{
                    "@odata.type" = "#microsoft.graph.aadUserConversationMember"
                    Roles = @(
                        "owner"
                    )
                    "User@odata.bind" = "https://graph.microsoft.com/v1.0/users('"+(get-mguser -userid (Get-MgContext).account).id +"')"
                }
                @{
                    "@odata.type" = "#microsoft.graph.aadUserConversationMember"
                    Roles = @(
                        "owner"
                    )
                    "User@odata.bind" = "https://graph.microsoft.com/v1.0/users('"+$RecpID.id +"')"
                }
            )
        }

        $ChatSessionID = New-MgChat -BodyParameter $NewChatIDParam

        Write-Host "Sending Message to $($RecpID.Mail)" -ForegroundColor Green

        try {
            #### Sending The Message
            $Body = @{
                ContentType = 'html'
                Content = @"
                Hello $($RecpID.DisplayName)<br>
                Your password will expire in $($DaysToSendWarning), Please follow <Strong><a href='www.office.com'>the instruction here to update it</a> </Strong> <BR>
                Thanks for your attention
"@
              }

        New-MgChatMessage -ChatId $ChatSessionID.ID -Body $Body -Importance Urgent
        } catch{
            Write-Host $_.Exception.Message
        }
    }
}
```

**Review by michaeltlombardi — COMMENTED** [review]
Thank you so much for this submission, @farismalaeb!! This looks great! I made a few suggestions around styling and formatting that I think will help readers a little bit.

**Comment by farismalaeb** [comment]
@michaeltlombardi 
Thanks for your efforts. I appreciate it. :)
All are committed and ready
Thanks again.

File: Posts/2022/07/Password Expiry Notification Using Teams and Graph API [hunk 20]
```diff
@@ -0,0 +1,334 @@
...
+Microsoft had a different endpoint for each cloud service.
+This makes it hard for the admin as it needs knowledge of each endpoint API URI and manages the authentication and authorization separately.
+So Microsoft came up with the Graph API as a one-stop shop to manage all the cloud services using a single endpoint, authentication, and a scoped authorization.
+
+[Microsoft Graph](https://docs.microsoft.com/en-us/graph/overview) is the gateway to read data from a wide range of Microsoft services, including Azure Active Directory, Teams, OneDrive…etc.
+You can get the data using a single module and a single interface.
+
+Microsoft Graph API supports modern authentication protocols such as access token, certificate, and browser authentication.
+
+> You can read more about the Graph API available endpoint from the [Microsoft Graph REST API Endpoint v1.0 Reference](https://docs.microsoft.com/en-us/graph/api/overview?view=graph-rest-1.0&preserve-view=true).
```
Comment (sdwheeler) [inline Posts/2022/07/Password Expiry Notification Using Teams and Graph API:26]:
```suggestion
> You can read more about the Graph API available endpoint from the [Microsoft Graph REST API Endpoint v1.0 Reference](https://docs.microsoft.com/graph/api/overview).
```

File: Posts/2022/07/Password Expiry Notification Using Teams and Graph API [hunk 21]
```diff
@@ -0,0 +1,334 @@
...
+Later on, the admin wants to add the `User.ReadWrite.All` permission, all the admin needs to do is run the `Connect-MgGraph` and set the new scope
+
+```powershell
+$Scope=@('User.ReadWrite.All')
+Connect-MgGraph -Scopes $Scope
+```
+
+The new scope permission adds to the current one.
+
+A good starting point in finding out the required permission to execute a certain cmdlet is [Find-MgGraphcommand](https://docs.microsoft.com/en-us/powershell/microsoftgraph/find-mg-graph-command?view=graph-powershell-1.0) and [Graph Explorer](https://developer.microsoft.com/en-us/graph/graph-explorer).
```
Comment (sdwheeler) [inline Posts/2022/07/Password Expiry Notification Using Teams and Graph API:96]:
```suggestion
A good starting point in finding out the required permission to execute a certain cmdlet is [Find-MgGraphcommand](https://docs.microsoft.com/powershell/microsoftgraph/find-mg-graph-command) and [Graph Explorer](https://developer.microsoft.com/graph/graph-explorer).
```

File: Posts/2022/07/Password Expiry Notification Using Teams and Graph API [hunk 22]
```diff
@@ -0,0 +1,334 @@
...
+### Getting Password Expiration information
+
+To make this tutorial more fun, let's make a scenario.
+Consider a company named Contoso.com.
+Contoso.com have an on-premise AD syncing with AAD. The Password Expiration policy is set to 3 months.
+
+The administrator is looking for a way to send the users a notification through Microsoft Teams chat one week before the password expires, so, how to start?
+
+To get the password expiration for users, use the following code.
+This code reads the **Name**, **EmailAddress**, **UserPrincipalName** and [**msDS-UserPasswordExpiryTimeComputed**](https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-adts/f9e9b7e2-c7ac-4db6-ba38-71d9696981e9).
```
Comment (sdwheeler) [inline Posts/2022/07/Password Expiry Notification Using Teams and Graph API:109]:
```suggestion
This code reads the **Name**, **EmailAddress**, **UserPrincipalName** and [**msDS-UserPasswordExpiryTimeComputed**](https://docs.microsoft.com/openspecs/windows_protocols/ms-adts/f9e9b7e2-c7ac-4db6-ba38-71d9696981e9).
```

File: Posts/2022/07/Password Expiry Notification Using Teams and Graph API [hunk 23]
```diff
@@ -0,0 +1,334 @@
...
+.
+Output trimmed
+```
+
+Later on a `ForEach-Object` loop goes through the users, and if any user **msDS-UserPasswordExpiryTimeComputed** matches the date in the `$DaysToSendWarning`, then the script sends them the chat message.
+More to come in the full script.
+
+## Using Microsoft Graph to Create a Teams Chat Session
+
+[Microsoft Graph API documentation](https://docs.microsoft.com/en-us/graph/) is the best start with anything related to Graph API.
```
Comment (sdwheeler) [inline Posts/2022/07/Password Expiry Notification Using Teams and Graph API:164]:
```suggestion
[Microsoft Graph API documentation](https://docs.microsoft.com/graph/) is the best start with anything related to Graph API.
```

File: Posts/2022/07/Password Expiry Notification Using Teams and Graph API [hunk 24]
```diff
@@ -0,0 +1,334 @@
...
+19:b980153c-9129-9129-9129-fb57a348d4d3_eafa5e65-9129-4e69-4e69-19f3fe6@unq.gbl.spaces
+
+```
+
+Replace each `*XXXXXXXXXX*` with a user ID who is participating in the chat. It doesn't matter if the sender or the recipient is first. it's a two-way communication bridge.
+But that the caller user id must be one of the members specified in the request body.
+
+You can get the user id by running `(Get-MgUser -userID user1@contoso.com).id`
+
+Read more about the parameters in the chat session from the [Create chat](https://docs.microsoft.com/en-us/graph/api/chat-post?view=graph-rest-beta&tabs=http).
```
Comment (sdwheeler) [inline Posts/2022/07/Password Expiry Notification Using Teams and Graph API:209]:
```suggestion
Read more about the parameters in the chat session from the [Create chat](https://docs.microsoft.com/graph/api/chat-post?view=graph-rest-beta&tabs=http).
```

File: Posts/2022/07/Password Expiry Notification Using Teams and Graph API [hunk 25]
```diff
@@ -0,0 +1,334 @@
...
+## Using Microsoft Graph to Send a Teams Chat Message
+
+As for now, we have the chat session id.
+We can send a message with a line of code.
+
+```powershell
+New-MgChatMessage -ChatId $ChatSessionID.id -Body  @{Content ='<strong>Hello, I am PowerShell</strong>';ContentType='html'}
+```
+
+Looking to know more about `New-MgChatMessage` parameters, take a look at [Send chatMessage in channel or a chat](https://docs.microsoft.com/en-us/graph/api/chatmessage-post?view=graph-rest-beta&tabs=http) and also [Send HTML Teams Message Using PowerShell Graph](https://www.powershellcenter.com/2022/07/15/new-mgchat/).
```
Comment (sdwheeler) [inline Posts/2022/07/Password Expiry Notification Using Teams and Graph API:225]:
```suggestion
Looking to know more about `New-MgChatMessage` parameters, take a look at [Send chatMessage in channel or a chat](https://docs.microsoft.com/graph/api/chatmessage-post?view=graph-rest-beta&tabs=http) and also [Send HTML Teams Message Using PowerShell Graph](https://www.powershellcenter.com/2022/07/15/new-mgchat/).
```

File: Posts/2022/07/Password Expiry Notification Using Teams and Graph API [hunk 26]
```diff
@@ -0,0 +1,334 @@
...
+
+```powershell
+Install-Module -Name Microsoft.Graph
+```
+
+Microsoft PowerShell Graph Module SDK is cross-platform and supports Windows, macOS, and Linux.
+
+## Connecting To Graph API Using PowerShell
+
+Unlike other modules such as the AzureAD, ExchangeOnline…etc.
```
Comment (sdwheeler) [inline Posts/2022/07/Password Expiry Notification Using Teams and Graph API:40]:
```suggestion
Unlike other modules such as the AzureAD, ExchangeOnline, etc.
```

File: Posts/2022/07/Password Expiry Notification Using Teams and Graph API [hunk 27]
```diff
@@ -0,0 +1,334 @@
...
+
+But first, let's talk about Graph API, so we are all on the same page.
+
+## What is the Graph API?
+
+Microsoft had a different endpoint for each cloud service.
+This makes it hard for the admin as it needs knowledge of each endpoint API URI and manages the authentication and authorization separately.
+So Microsoft came up with the Graph API as a one-stop shop to manage all the cloud services using a single endpoint, authentication, and a scoped authorization.
+
+[Microsoft Graph](https://docs.microsoft.com/en-us/graph/overview) is the gateway to read data from a wide range of Microsoft services, including Azure Active Directory, Teams, OneDrive…etc.
```
Comment (sdwheeler) [inline Posts/2022/07/Password Expiry Notification Using Teams and Graph API:21]:
```suggestion
[Microsoft Graph](https://docs.microsoft.com/graph/overview) is the gateway to read data from a wide range of Microsoft services, including Azure Active Directory, Teams, OneDrive…etc.
```

**Pull request merged**